### PR TITLE
Allow configuring falcon output dir

### PR DESCRIPTION
**Pull request by andrewjstone**

### CODE REVIEW

File: lib/src/lib.rs [hunk 1]
```diff
@@ -60,6 +60,8 @@ macro_rules! cmd {
     }};
 }
 
+pub const DEFAULT_FALCON_DIR: &str = ".falcon";
```
Comment (andrewjstone) [inline lib/src/lib.rs:63]: I didn't use `$pwd` here, because it doesn't fit in a const as an environment variable. It also doesn't seem necessary.

**Comment by andrewjstone** [comment]
I tested this by launching a `solo` example vm and logging in with serial, running exec, etc... I also changed the test vm launcher in omicron to use tempdirs. I then left them running on failure and logged in after the fact, then destroyed them via CLI.

File: lib/src/cli.rs [hunk 2]
```diff
@@ -78,28 +79,48 @@ struct CmdLaunch {
...
+    falcon_dir: Utf8PathBuf,
 }
 
 #[derive(Parser)]
 #[clap(infer_subcommands = true)]
-struct CmdPreflight {}
+struct CmdPreflight {
+    /// The path of the falcon output directory
+    #[clap(short, long, default_value_t = Utf8PathBuf::from(DEFAULT_FALCON_DIR))]
+    falcon_dir: Utf8PathBuf,
```
Comment (rcgoodfellow) [inline lib/src/cli.rs:93]: Is plumbing these options into the commands like this so you can run one precompiled program in multiple directories?

**Comment by rcgoodfellow** [comment]
Delightfully, this PR makes this readme line go away

https://github.com/oxidecomputer/falcon/blob/a87513f83ed3b4ce84aaf91f3e0d49306dde386b/README.md?plain=1#L111-L112

Perhaps replace this with a bit of guidance on using the new configurability.

File: lib/src/cli.rs (hunk 2 above)
Comment (andrewjstone) [reply lib/src/cli.rs:93]: Basically, yes. I want to be able to run a test inside a VM and then leave it running if it fails. And if so, I want to be able to attach via CLI and debug.

**Comment by andrewjstone** [comment]
> Delightfully, this PR makes this readme line go away
> 
> https://github.com/oxidecomputer/falcon/blob/a87513f83ed3b4ce84aaf91f3e0d49306dde386b/README.md?plain=1#L111-L112
> 
> Perhaps replace this with a bit of guidance on using the new configurability.

Good idea!

File: lib/src/cli.rs (hunk 2 above)
Comment (andrewjstone) [reply lib/src/cli.rs:93]: I plumbed them into individual commands just to match the existing pattern. It could probably be a top level option.

**Comment by andrewjstone** [comment]
> Delightfully, this PR makes this readme line go away
> 
> https://github.com/oxidecomputer/falcon/blob/a87513f83ed3b4ce84aaf91f3e0d49306dde386b/README.md?plain=1#L111-L112
> 
> Perhaps replace this with a bit of guidance on using the new configurability.

Done.

**Review by rcgoodfellow — APPROVED** [review]
Looks good to me and works with `a4x2`. Thanks!

**Comment by andrewjstone** [comment]
> Looks good to me and works with `a4x2`. Thanks!

Thanks for the quick review again Ry!

**Pull request merged**